src/pg_to_evalscript/javascript_processes/variance.js

```diff
@@ -0,0 +1,50 @@
+function variance(arguments) {
+  const { data, ignore_nodata = true } = arguments;
+
+  if (data === undefined) {
+    throw new Error("Mandatory argument `data` is not defined.");
+  }
+
+  if (!Array.isArray(data)) {
+    throw new Error("Argument `data` is not an array.");
+  }
+
+  if (typeof ignore_nodata !== "boolean") {
+    throw new Error("Argument `ignore_nodata` is not a boolean.");
+  }
+
+  let sum = 0;
+  let count = 0;
+  for (let x of data) {
+    if (typeof x !== "number" && x !== null) {
+      throw new Error("Value in argument `data` is not a number or null.");
+    }
+
+    if (x === null) {
+      if (!ignore_nodata) {
+        return null;
+      } else {
+        continue;
+      }
+    }
+
+    sum += x;
+    count++;
+  }
+
+  if (count === 0) {
+    return null;
+  }
+
+  const mean = sum / count;
+  let sumOfSquares = 0;
+  for (let x of data) {
+    if (ignore_nodata && x === null) {
+      continue;
+    }
+
+    sumOfSquares += Math.pow(x - mean, 2);
+  }
+
+  return sumOfSquares / (count - 1);
```

Comment:
Mostly because the formula used for the examples in the docs is the one for "bias-corrected sample variance" (formula 5 in https://mathworld.wolfram.com/Variance.html)

It's the same in `sd` process https://github.com/openEOPlatform/openeo-pg-evalscript-converter/pull/42
(formula 5 in https://mathworld.wolfram.com/StandardDeviation.html)

Reply:
I think you're right, although it is not obvious from definition of process. But I see that https://github.com/Open-EO/openeo-processes-python/blob/master/src/openeo_processes/math.py also uses n-1 (ddof=1 ).

Reply:
Handled the additional use case where there is only 1 element in the `data` array. In this case I just return 0, since the upper part of the fraction in the formula (the sum) will always be 0.